workflow/controller/operator.go

```diff
@@ -1444,10 +1444,10 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, old *wfv1.NodeStatus
 		}
 	}
 
-	// We cannot fail the node until the wait container is finished (unless any init container has failed) because it may be busy saving outputs, and these
+	// We cannot fail the node until the wait container is finished (unless any init container has failed or wait contain is still Pending) because it may be busy saving outputs, and these
 	// would not get captured successfully.
 	for _, c := range pod.Status.ContainerStatuses {
-		if (c.Name == common.WaitContainerName && c.State.Terminated == nil && new.Phase.Completed()) && !initContainerFailed {
+		if (c.Name == common.WaitContainerName && c.State.Terminated == nil && c.State.Waiting == nil && new.Phase.Completed()) && !initContainerFailed {
```

Comment:
this could probably be simplified to change `c.State.Terminated == nil && c.State.Waiting == nil` to `c.State.Running != nil`, right?

Reply:
I think this is okay